src/hyperlane-adapter/HyperlaneSenderAdapter.sol

```diff
@@ -0,0 +1,293 @@
+
+  /**
+   * @notice Emitted when a domain identifier for a destination chain is updated.
+   * @param dstChainId Destination chain identifier.
+   * @param dstDomainId Destination domain identifier.
+   */
+  event DestinationDomainUpdated(uint256 dstChainId, uint32 dstDomainId);
+
+  /**
+   * @notice HyperlaneSenderAdapter constructor.
+   * @param _mailbox Address of the Hyperlane `Mailbox` contract.
+   */
+  constructor(address _mailbox, address _igp) {
+    if (_mailbox == address(0)) {
+      revert Errors.InvalidMailboxZeroAddress();
+    }
+    mailbox = IMailbox(_mailbox);
+    _setIgp(_igp);
+  }
+
+  /// @dev we narrow mutability (from view to pure) to remove compiler warnings.
+  /// @dev unused parameters are added as comments for legibility.
+  function getMessageFee(
+    uint256 toChainId,
+    address,
+    /* to*/
+    bytes calldata /* data*/
+  ) external view returns (uint256) {
+    uint32 dstDomainId = _getDestinationDomain(toChainId);
+    // destination gasAmount is hardcoded to 500k similar to Wormhole implementation
+    // See https://docs.hyperlane.xyz/docs/build-with-hyperlane/guides/paying-for-interchain-gas
+    try igp.quoteGasPayment(dstDomainId, 500000) returns (uint256 gasQuote) {
+      return gasQuote;
+    } catch {
+      // Default to zero, MultiMessageSender.estimateTotalMessageFee doesn't expect this function to revert
+      return 0;
+    }
+  }
+
+  /**
+   * @notice Sets the IGP for this adapter.
+   * @dev See _setIgp.
+   */
+  function setIgp(address _igp) external onlyOwner {
+    _setIgp(_igp);
+  }
+
+  function dispatchMessage(
+    uint256 _toChainId,
+    address _to,
+    bytes calldata _data
+  ) external payable override returns (bytes32) {
+    IMessageExecutor adapter = _getMessageExecutorAddress(_toChainId);
+    _checkAdapter(_toChainId, adapter);
+    uint32 dstDomainId = _getDestinationDomain(_toChainId);
+
+    if (dstDomainId == 0) {
+      revert Errors.UnknownDomainId(_toChainId);
+    }
+
+    uint256 _nonce = _incrementNonce();
+    bytes32 msgId = MessageLib.computeMessageId(_nonce, msg.sender, _to, _data);
+
+    MessageLib.Message[] memory _messages = new MessageLib.Message[](1);
+    _messages[0] = MessageLib.Message({ to: _to, data: _data });
+
+    bytes memory payload = abi.encode(_messages, msgId, block.chainid, msg.sender);
+
+    bytes32 hyperlaneMsgId = IMailbox(mailbox).dispatch(
+      dstDomainId,
+      TypeCasts.addressToBytes32(address(adapter)), //receiver adapter is the reciever
+      // Include the source chain id so that the receiver doesn't have to maintain a srcDomainId => srcChainId mapping
+      payload
+    );
+
+    // try to make gas payment, ignore failures
+    // destination gasAmount is hardcoded to 500k similar to Wormhole implementation
+    // refundAddress is set from MMS caller state variable
+    try
+      igp.payForGas{ value: msg.value }(
+        hyperlaneMsgId,
+        dstDomainId,
+        500000,
```

Comment:
Move to constant/storage

Reply:
[Moved gasAmount to storage](https://github.com/pooltogether/ERC5164/pull/17/commits/9bdd08bf464430ce67042308eac41cfd2865fee5#r1385047124)